core/src/main/scala/com/salesforce/op/ModelInsights.scala

```diff
@@ -697,8 +698,41 @@ case object ModelInsights {
           val metrics = rawFeatureFilterResults.rawFeatureFilterMetrics.filter(_.name == fname)
           val distributions = rawFeatureFilterResults.rawFeatureDistributions.filter(_.name == fname)
           val exclusionReasons = rawFeatureFilterResults.exclusionReasons.filter(_.name == fname)
-          FeatureInsights(featureName = fname, featureType = ftype, derivedFeatures = seq.map(_._2),
-            metrics = metrics, distributions = distributions, exclusionReasons = exclusionReasons)
+          val sensitiveFeatureInformation = vectorInfo.flatMap(_.sensitive.get(fname)) match {
+            case Some(info) => info
+            case _ => Seq.empty
+          }
+          FeatureInsights(
+            featureName = fname, featureType = ftype, derivedFeatures = seq.map(_._2),
+            metrics = metrics, distributions = distributions, exclusionReasons = exclusionReasons,
+            sensitiveInformation = sensitiveFeatureInformation
+          )
+      }.toSeq ++ {
+        /*
+          Add FeatureInsights for removed sensitive fields that do not have a column in OpVectorMetadata.
+          With current TMOG settings, this will not happen unless null tracking is turned off since
+          null indicators are created for all text features, even ignored ones.
+        */
+        vectorInfo match {
```

Comment:
Hmmm, not a fan of having to do this. To make sure I understand, this is necessary because we determine ignorability (when not to hash or pivot) in the SmartTextVectorizer but determine the sensitive feature info in a separate estimator?

I thought the sensitive feature information was on a raw feature level, not a derived feature level. If so, then wouldn't sensitive feature info already be decoupled from the size of the derived features? Am I misunderstanding this?

Reply:
he sensitive feature information _is_ on the raw feature level _but_ in my earlier testing (when I was completely dropping any derived features for name columns, including null and length trackers), I found that `FeatureInsights` are only created for raw features that have derived features. I can probably(?) take this code out now though, since we've agreed that the `Ignore` STV method will still generate null/length tracker columns, so long as those features are enabled.

If I take the code out though, there is the situation that if we decide to turn off null/length tracking _and_ some fields are ignored, then sensitive feature information won't be generated for those raw features. But maybe this situation is unlikely enough that it's fine to remove this bit of code.

Reply:
Hmmm, I guess we should leave it in. It's not hurting anything, I'm just unhappy with our inconsistent treatment of when we decide to remove features.